### PR TITLE
docs: Update roadmap for 1.13

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The roadmap doc is updated with some high level features that are targeted for the 1.13 release.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by subhamkrai — COMMENTED** [review]
LGTM

File: ROADMAP.md [hunk 1]
```diff
@@ -8,36 +8,23 @@ We hope that the items listed below will inspire further engagement from the com
 Any dates listed below and the specific issues that will ship in a given milestone are subject to change but should give a general idea of what we are planning.
 See the [GitHub project boards](https://github.com/rook/rook/projects) for the most up-to-date issues and their status.
 
-## Rook Ceph 1.12
+## Rook Ceph 1.13
 
-The following high level features are targeted for Rook v1.12 (July 2023). For more detailed project tracking see the [v1.12 board](https://github.com/rook/rook/projects/29).
+The following high level features are targeted for Rook v1.13 (November 2023). For more detailed project tracking see the [v1.13 board](https://github.com/rook/rook/projects/30).
```
Comment (obnoxxx) [inline ROADMAP.md:13]:
```suggestion
The following high level features are targeted for Rook v1.13 (November 2023). For more detailed project tracking see the [v1.13 board(https://github.com/rook/rook/projects/30).
```

Comment (travisn) [reply ROADMAP.md:13]: The link looks correct to me with the `]`

Comment (obnoxxx) [reply ROADMAP.md:13]: @travisn  wrote:
> The link looks correct to me with the `]`

right, my suggestion was meant to be different, just intended to remove the space between `]` and `(` . I fixed the suggestion accordingly ...

Comment (travisn) [reply ROADMAP.md:13]: hmm I don't see a space. Are you just looking at the github UI adding a newline there for the view?

Comment (obnoxxx) [reply ROADMAP.md:13]: @travisn  wrote:
> hmm I don't see a space. Are you just looking at the github UI adding a newline there for the view?


quite possible.
if there isn't a newline in the patch, then maybe some other whitespace which I was suggesting  to remove

Comment (travisn) [reply ROADMAP.md:13]: If I expand my browser, it does not wrap or show any space there

Comment (obnoxxx) [reply ROADMAP.md:13]: I  checked it by locally fetching this PR. there is no whitespace. the link formatting is clean!

**Review by obnoxxx — APPROVED** [review]
LGTM

**Review by satoru-takeuchi — APPROVED** [review]

File: ROADMAP.md [hunk 2]
```diff
@@ -8,36 +8,23 @@ We hope that the items listed below will inspire further engagement from the com
...
 
-* Automate node fencing for application failover in some scenarios [#1507](https://github.com/rook/rook/issues/1507)
 * OSD encryption on partitions [#10984](https://github.com/rook/rook/issues/10984)
-* Support IPv6 for external clusters [#11602](https://github.com/rook/rook/issues/11602)
 * Object Store
-  * Add alpha support for COSI (Container object storage interface) with K8s 1.25 [#7843](https://github.com/rook/rook/issues/7843)
-  * Support RGW LDAP integration [#4315](https://github.com/rook/rook/issues/4315)
-  * Service account authentication with the Vault agent [#9872](https://github.com/rook/rook/pull/9872)
+  * Declare COSI stable
```
Comment (leseb) [inline ROADMAP.md:17]: 🥳

**Review by leseb — APPROVED** [review]

**Pull request merged**